From f851c033ccfae1d6ed4bc39bd147d56fb839f8d3 Mon Sep 17 00:00:00 2001
From: Hieu Huynh <hieuh@os.amperecomputing.com>
Date: Tue, 18 Oct 2022 07:02:11 +0000
Subject: [PATCH] Add Hostinterface CredentialBootstrapping interface

Adds the Hostinterface CredentialBootstrapping interface
xyz.openbmc_project.HostInterface.CredentialBootstrapping to support the
HostInterface schema.

Upstream-Status: Pending

Signed-off-by: Hieu Huynh <hieuh@os.amperecomputing.com>
---
 users.hpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/users.hpp b/users.hpp
index 77a1079..13e98f0 100644
--- a/users.hpp
+++ b/users.hpp
@@ -18,6 +18,7 @@
 #include <sdbusplus/server/object.hpp>
 #include <xyz/openbmc_project/Object/Delete/server.hpp>
 #include <xyz/openbmc_project/User/Attributes/server.hpp>
+#include <xyz/openbmc_project/HostInterface/CredentialBootstrapping/server.hpp>

 namespace phosphor
 {
@@ -27,7 +28,8 @@ namespace user
 namespace Base = sdbusplus::xyz::openbmc_project;
 using UsersIface = Base::User::server::Attributes;
 using DeleteIface = Base::Object::server::Delete;
-using Interfaces = sdbusplus::server::object_t<UsersIface, DeleteIface>;
+using HostInterface = Base::HostInterface::server::CredentialBootstrapping;
+using Interfaces = sdbusplus::server::object_t<UsersIface, DeleteIface, HostInterface>;
 // Place where all user objects has to be created
 constexpr auto usersObjPath = "/xyz/openbmc_project/user";

--
2.25.1

